pwiz_tools/Skyline/Controls/Graphs/SummaryRelativeAbundanceGraphPane.cs

```diff
@@ -158,7 +158,13 @@ private void UpdateFormatting(IEnumerable<MatchRgbHexColor> colorRows)
         public void OnLabelOverlapPropertyChange(object sender, PropertyChangedEventArgs e)
         {
             if (e.PropertyName == @"GroupComparisonAvoidLabelOverlap")
+            {
+                Settings.Default.PropertyChanged -= OnLabelOverlapPropertyChange;
+                Settings.Default.GroupComparisonSuspendLabelLayout = false;
+                _labelsLayout = null;
                 GraphSummary.UpdateUI();
+                Settings.Default.PropertyChanged += OnLabelOverlapPropertyChange;
```

Comment:
If I have to write code which removes a listener and adds it back I usually put it inside of a try/finally:
```
try 
{
    Settings.Default.PropertyChanged -= OnLabelOverlapPropertyChange;
// do other stuff
}
finally
{
    Settings.Default.PropertyChanged += OnLabelOverlapPropertyChange;
}
```
That way, even if some sort of exception happens, things still get back into an expected state.

For something like this, I would probably just have a field:
`private bool _inPropertyChange;
`
and then in the property change handler:
```
if (_inPropertyChange) 
{
   return;
}
try 
{
    _inPropertyChange = true;
    if (e.PorpertyName == nameof(Settings.Default.GroupComparisonAvoidLabelOverlap)) 
    {
        // do some stuff
    }
    else if (e.PropertyName == nameof(Settings.Default.GroupComparisonSuspendLabelLayout)) 
    {
        // do other stuff
    }
}
finally 
{
    _inPropertyChange = false;
}
```
But the way you have this is fine too.